### PR TITLE
[ADD] r_planning: filter for resource allocation 

### PR DESCRIPTION
**Pull request by remytms**
This add a filter to show only resource allocation that are reserved for
maintenance. This also add a filter to show only resource allocation
with begin date <= today <= end date.

Warning, to easily review, read only commit d12351f.

### CODE REVIEW

**Comment by vdewulf** [comment]
@robinkeunen could you review please ? Thanks!

File: resource_planning/views/resource_planning_views.xml [hunk 1]
```diff
@@ -154,6 +154,11 @@
           <field name="location" string="location" groups="resource_planning.group_multi_location"/>
           <filter string="Option" name="option" domain="[('state','=','option')]"/>
           <filter string="Booked" name="booked" domain="[('state','=','booked')]"/>
+          <filter string="Maintenance" name="maintenance" domain="[('state','=','maintenance')]"/>
+          <separator/>
+          <filter string="Currently" name="currently"
+            domain="[('date_start','&lt;=', datetime.datetime.combine(context_today(), datetime.time(23, 59, 59))),
```
Comment (robinkeunen) [inline resource_planning/views/resource_planning_views.xml]: This does not do what the commit message says and I don't think it is correct.

If the resource is booked from today 10:00 to today 18:00, it won't show up in the _currently_ filter.

**Review by robinkeunen — CHANGES_REQUESTED** [review]
The maintenance filter is ok but I think the _Currently_ filter is wrong. It should be  `start < now* < end`.

* Now as in current time and not current day.

File: resource_planning/views/resource_planning_views.xml (hunk 1 above)
Comment (remytms) [reply resource_planning/views/resource_planning_views.xml]: So let suppose resource booked as you said : `date_start = today 10:00` and `date_end = today 18:00`.

The domain says : `date_start <= today 23:59:59 AND date_end >= today 00:00:00`, so the resource will show up.

**Comment by remytms** [comment]
@robinkeunen ,

C'est une interprétation différente. Dans la tâche il est indiqué : 

> Ajouter un filtre prédéfini (non mis par défaut mais facilement accessible en cliquant sur les filtres) indiquant :
> - date de début de réservation =< aujourd'hui =< date de fin de réservation

Donc j'ai fait en sorte que ça joue sur les dates  et non sur les heures. Qu'en penses-tu ?

File: resource_planning/views/resource_planning_views.xml (hunk 1 above)
Comment (robinkeunen) [reply resource_planning/views/resource_planning_views.xml]: Yup, my bad. :-)

**Comment by robinkeunen** [comment]
Avec ma nouvelle compréhension du filtre, je pense que c'est bon :+1:

**Review by robinkeunen — APPROVED** [review]

**Comment by remytms** [comment]
Merci pour ta relecture attentive. :)

**Pull request merged**